server/src/main/java/io/crate/metadata/information/InformationSchemaTableDefinitions.java

```diff
@@ -67,9 +67,10 @@ public InformationSchemaTableDefinitions(InformationSchemaIterables informationS
         ));
         tableDefinitions.put(InformationColumnsTableInfo.IDENT, new StaticTableDefinition<>(
             informationSchemaIterables::columns,
-            (user, c) -> user.hasAnyPrivilege(Privilege.Clazz.TABLE, c.relation().ident().fqn())
+            (user, c) -> (user.hasAnyPrivilege(Privilege.Clazz.TABLE, c.relation().ident().fqn())
                          // we also need to check for views which have privileges set
-                         || user.hasAnyPrivilege(Privilege.Clazz.VIEW, c.relation().ident().fqn()),
+                         || user.hasAnyPrivilege(Privilege.Clazz.VIEW, c.relation().ident().fqn())
+                         ) && !c.ref().isDropped(), // Align with PG behavior where dropped columns are shown only in pg_attribute table.
```

Comment:
I'd say that this is default behaviour and SQL standard (not completely sure and I cannot find any mentioning of dropped columns in the related SQL standard) and showing dropped columns in `pg_attribute` is special and aligns with PG.
```suggestion
                         ) && !c.ref().isDropped(),
```